### PR TITLE
Temporarybranch

### PR DESCRIPTION
**Pull request by ENguyen25**

### CODE REVIEW

File: .env.EXAMPLE [hunk 1]
```diff
@@ -1,3 +0,0 @@
-DB_NAME=ubiquitous_db
-DB_USER=root
-DB_PASSWORD=
```
Comment (Myuze) [inline .env.EXAMPLE:3]: You deleted the .env.EXAMPLE file again? ^^;

File: controllers/api/user-routes.js [hunk 2]
```diff
@@ -25,6 +25,40 @@ router.post('/register', async (req, res) => {
   }
 });
 
+// gets one user by id
+router.get('/:id', async (req, res) => {
+  const { user } = req.session;
+
+  // if (user){
```
Comment (Myuze) [inline controllers/api/user-routes.js:32]: Should just remove this if it's not going to be used.

File: controllers/api/user-routes.js [hunk 3]
```diff
@@ -25,6 +25,40 @@ router.post('/register', async (req, res) => {
...
+    const findUser = await User.findOne({
+      where: {
+          id: req.params.id,
+          include: {
+            model: 'User',
+            attributes: [ 'id', 'username' ]
+          }
+      }
```
Comment (Myuze) [inline controllers/api/user-routes.js:40]:
```suggestion
  const findUser = await User.findOne({
    where: {
        id: req.params.id,
        include: {
          model: 'User',
          attributes: [ 'id', 'username' ]
        }
    }
```

File: controllers/user-routes.js [hunk 4]
```diff
@@ -52,19 +52,41 @@ router.get('/newPost', async (req, res) => {
...
-            .status(400)
-            .render('new-post', { message: 'Please login or register to make a new post.'});
+            .status(200)
+            .render('add-post', { message: 'Please login or register to make a new post.'})
         }
     } catch (err) {
       console.log(err);
         res.status(500).json(err);
     }
   });
```
Comment (Myuze) [inline controllers/user-routes.js:67]: It's better practice to check for the negative status first, also "else" already counts for the inverse of "user", so another "if" is not necessary.
`        if (!user) {
            res
            .status(200)
            .render('add-post', { message: 'Please login or register to make a new post.'})
        } else {
           res
            .status(200)
            .render('add-post') 
        }
    } catch (err) {
      console.log(err);
        res.status(500).json(err);
    }`

File: controllers/user-routes.js [hunk 5]
```diff
@@ -52,19 +52,41 @@ router.get('/newPost', async (req, res) => {
...
+//post route for making a new post inserting into db
+router.post('/newPost', async (req, res) => {
+  const user = req.session
+  try {
+    if (user){
+      res
+        .status(400)
+        .redirect('/login', {message: 'please login to make a new post'})
+    }
+    if (!user){
```
Comment (Myuze) [inline controllers/user-routes.js:78]:
```suggestion
    } else {
```

Comment (Myuze) [reply controllers/user-routes.js:78]: I suggest checking for "if (!user)" first.

File: models/Comment.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        key: 'id',
+      },
+    }
+  },
+  {
+    sequelize,
+    timestamps: false,
+    freezeTableName: true,
+    underscored: true,
+    modelName: 'comments',
```
Comment (Myuze) [inline models/Comment.js:50]:
```suggestion
    modelName: 'comment',
```

File: public/js/new-post.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    if (response.ok) {
+      document.location.replace('/forum');
+    } else {
+      alert('Failed to add post');
+    }
+  }
+
+  document
+    .querySelector('.new-forum-post')
+    .addEventListener('submit', newFormHandler);
```
Comment (Myuze) [inline public/js/new-post.js:28]:
```suggestion
    .addEventListener('submit', newFormHandler);
    
```

File: controllers/user-routes.js [hunk 8]
```diff
@@ -52,19 +52,41 @@ router.get('/newPost', async (req, res) => {
...
+        entry: req.body.entry
+    })
+      res
+        .status(200)
+        .json(dbPostData)
+    }
+  } catch (err) {
+    console.log(err)
+  }
+})
```
Comment (Myuze) [inline controllers/user-routes.js:90]: I don't think this logic does what you think,
Line 73: If there is a "user session" please login?
Line 78: If there is a NOT a "user session" post an entry?
Seems backwards to me.

File: models/UserComment [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+  {
+    sequelize,
+    timestamps: false,
+    freezeTableName: true,
+    underscored: true,
+    modelName: 'comments',
+  }
+);
+
+module.exports = UserComment;
```
Comment (Myuze) [inline models/UserComment:52]: For the .js extension for the file.tt

File: models/index.js [hunk 10]
```diff
@@ -26,6 +27,36 @@ Data.belongsTo(User, {
...
+      unique: false
+    }
+})
+
+Comment.belongsTo(Post, {
+    through: {
+      model: User,
+      unique: false
+    }
+})
```
Comment (Myuze) [inline models/index.js:58]: The "through" property only works with belongsToMany releation ship in M:M or M:N.
[https://sequelize.org/master/manual/advanced-many-to-many.html](url)

**Review by Myuze — CHANGES_REQUESTED** [review]
Along with Michael's commits we need to discuss how the login flow works, and the proper way to validate before performing certain actions.

File: public/js/new-comment.js [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+      console.log('succeed to add post')
+    } else {
+      alert('Failed to add post');
+    }
+  }
+
+  document
+    .querySelector('.new-comment')
+    .addEventListener('submit', newCommentHandler);
+
```
Comment (Myuze) [inline public/js/new-comment.js:29]:
```suggestion
    .addEventListener('submit', newCommentHandler);

```

File: public/js/new-post.js [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+
```
Comment (Myuze) [inline public/js/new-post.js:28]:
```suggestion
    .addEventListener('submit', newFormHandler);

```

**Review by LordLowdar — APPROVED** [review]
༼ つ ◕_◕ ༽つ

**Pull request merged**